action.yml

```diff
@@ -280,20 +266,34 @@ runs:
         CA_GH_ACTION_CONTAINER_APP_ENVIRONMENT_VARIABLES_ARG="--env-vars ${{ inputs.environmentVariables }}"
         echo "CA_GH_ACTION_CONTAINER_APP_ENVIRONMENT_VARIABLES_ARG=${CA_GH_ACTION_CONTAINER_APP_ENVIRONMENT_VARIABLES_ARG}" >> $GITHUB_ENV
 
+    - name: Install pack CLI on non-Windows runner
+      if: ${{ runner.os != 'Windows' && inputs.appSourcePath != '' && env.CA_GH_ACTION_DOCKERFILE_PATH == '' }}
+      shell: bash
+      run: (curl -sSL "https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-${{ runner.os }}.tgz" | sudo tar -C /usr/local/bin/ --no-same-owner -xzv pack)
+
+    - name: Install pack CLI on Windows runner
+      if: ${{ runner.os == 'Windows' && inputs.appSourcePath != '' && && env.CA_GH_ACTION_DOCKERFILE_PATH == '' }}
+      shell: bash
+      run: |
+          mkdir -p $PWD/pack && cd $PWD/pack
+          curl -sSL "https://github.com/buildpacks/pack/releases/download/v0.27.0/pack-v0.27.0-windows.zip" -o "pack-windows.zip"
+          7z x pack-windows.zip > /dev/null 2>&1
+          rm pack-windows.zip && echo "$PWD" >> $GITHUB_PATH
+
     - name: Set Oryx++ Builder as default builder
       if: ${{ inputs.appSourcePath != '' && env.CA_GH_ACTION_DOCKERFILE_PATH == '' }}
       shell: bash
       run: pack config default-builder mcr.microsoft.com/oryx/builder:20230208.1
 
     - name: Set telemetry for Oryx++ Builder
-      if: ${{ inputs.disableTelemetry == 'false' }}
+      if: ${{ inputs.appSourcePath != '' && env.CA_GH_ACTION_DOCKERFILE_PATH == '' && inputs.disableTelemetry == 'false' }}
```

Comment:
I might misunderstand here but isn't the new condition going to make us send telemetry only if there is no Dockerfile? Am I missing something?

Reply:
This step (and the one below) are only relevant for the builder scenario -- we either are passing in an environment variable to the builder that says "don't collect telemetry" or an environment variable that says "here's the `CALLER_ID` for the `oryx build` call we're making in the builder". Telemetry will still be collected in the Dockerfile and previously built image cases.

Reply:
Maybe to be a bit more clear: this step and the one below determine the value of the environment variable `CA_GH_ACTION_ORYX_BUILDER_TELEMETRY_ARG`, which is passed along to the `pack build` command call and used nowhere else, scoping just to the builder scenario.

Reply:
Thanks for clarifying